client/asset/btc/btc.go

```diff
@@ -618,21 +653,115 @@ var _ asset.Wallet = (*baseWallet)(nil)
+	NumExternalAddresses uint32 `ini:"numinternaladdr"`
+	NumInternalAddresses uint32 `ini:"numexternaladdr"`
+}
+
+// GetRecoveryCfg returns information that will help the wallet get
+// back to its previous state after it is recreated. Part of the
+// Recoverer interface.
+func (btc *ExchangeWalletSPV) GetRecoveryCfg() (map[string]string, error) {
+	wallet, err := btc.nodeAccessor.getNode()
+	defer btc.nodeAccessor.returnNode()
+	if err != nil {
+		return nil, err
+	}
+	w := wallet.(*spvWallet)
+
+	internal, external, err := w.numDerivedAddresses()
+	if err != nil {
+		return nil, err
+	}
+
+	reCfg := &RecoveryCfg{
+		NumInternalAddresses: internal,
+		NumExternalAddresses: external,
+	}
+	cfg, err := config.Mapify(reCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+// ApplyRecoveryCfg gives the information returned from GetRecoveryCfg
+// to the new wallet. Part of the Recoverer interface.
+func (btc *ExchangeWalletSPV) ApplyRecoveryCfg(cfg map[string]string) error {
+	wallet, err := btc.nodeAccessor.getNode()
+	defer btc.nodeAccessor.returnNode()
+	if err != nil {
+		return err
+	}
+	w := wallet.(*spvWallet)
+
+	recoveryCfg := new(RecoveryCfg)
+	err = config.Unmapify(cfg, recoveryCfg)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < int(recoveryCfg.NumExternalAddresses); i++ {
+		_, err := btc.Address()
+		if err != nil {
+			return fmt.Errorf("unable to generate external address: %w", err)
+		}
+	}
+
+	for i := 0; i < int(recoveryCfg.NumInternalAddresses); i++ {
+		_, err := w.changeAddress()
+		if err != nil {
+			return fmt.Errorf("unable to generate change address: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Destroy will delete all the wallet files so the wallet can be recreated.
+// Part of the recoverer interface.
+func (btc *ExchangeWalletSPV) Destroy() error {
+	wallet := btc.nodeAccessor.startRecovery()
+	w := wallet.(*spvWallet)
+
+	err := w.deleteWalletData()
+	if err != nil {
+		return fmt.Errorf("unable to delete wallet data: %w", err)
+	}
+
+	return nil
+}
 
 // Rescan satisfies the asset.Rescanner interface, and issues a rescan wallet
 // command if the backend is an SPV wallet.
 func (btc *ExchangeWalletSPV) Rescan(_ context.Context) error {
 	// This will panic if not an spvWallet, which would indicate that
 	// openSPVWallet was not used to construct this instance.
-	w := btc.node.(*spvWallet)
+	wallet, err := btc.nodeAccessor.getNode()
+	defer btc.nodeAccessor.returnNode()
+	if err != nil {
+		return err
+	}
+	w := wallet.(*spvWallet)
 	atomic.StoreInt64(&btc.tipAtConnect, 0) // for progress
 	// Caller should start calling SyncStatus on a ticker.
 	return w.rescanWalletAsync()
 }
 
 // FeeRate satisfies asset.FeeRater.
 func (btc *ExchangeWalletFullNode) FeeRate() uint64 {
-	rate, err := btc.estimateFee(btc.node, 1)
+	wallet, err := btc.nodeAccessor.getNode()
+	defer btc.nodeAccessor.returnNode()
+	if err != nil {
+		btc.log.Error(err)
+		return 0
+	}
+	rate, err := btc.estimateFee(wallet, 1)
```

Comment:
Yeah there's no recovery in the `* ExchangeWalletFullNode`. Would it be bad to just leave it as it is and in the  `* ExchangeWalletFullNode` methods, just use `btc.nodeAccessor.node` instead of calling `getNode`?

Reply:
This is the only exclusively `* ExchangeWalletFullNode` node.